src/cli.js

```diff
@@ -0,0 +1,547 @@
+    report += 'With these changes, the following tests would fail:\n';
+    for (const test of testResults.fail) {
+      report += '<details><summary>${test.title}</summary>\n\n' +
+        'Expected:\n' +
+        '```json\n' +
+        JSON.stringify(test.expected, null, 2) +
+        '\n```\n' +
+        'Actual:\n' +
+        '```json\n' +
+        JSON.stringify(test.actual, null, 2) +
+        '\n```\n' +
+        '</details>\n';
+    }
+  }
+  else {
+    report += 'These changes look good! 😎\n';
+  }
+
+  return report;
+}
+
+
+/**
+ * Helper function to split a session issue body (in markdown) into sections
+ */
+function splitIssueBodyIntoSections(body) {
+  return body.split(/^### /m)
+    .filter(section => !!section)
+    .map(section => section.split(/\r?\n/))
+    .map(section => {
+      let value = section.slice(1).join('\n').trim();
+      if (value.replace(/^_(.*)_$/, '$1') === 'No response') {
+        value = null;
+      }
+      return {
+        title: section[0].replace(/ \(Optional\)$/, ''),
+        value
+      };
+    });
+}
+
+
+/*****************************************************************************
+ * Main loop, creates the CLI using Commander.
+ *****************************************************************************/
+const program = new Command();
+program
+  .name('browser-specs')
+  .version(version)
+  .description('Manage lists of specs in browser-specs: the main list, whose source is the `specs.json` file with resulting data in the `index.json` file, the monitor list (`src/data/monitor.json`) and the ignore list (`src/data/ignore.json`).');
+
+program
+  .command('build')
+  .summary('build the info for the given spec or list of changes made to `specs.json`.')
+  .description(`Build the info for the given spec or list of changes made to \`specs.json\`.`)
+  .argument('<what>', 'what to build. The argument may either be a `<url>` that represents the canonical URL of the spec to build, a named commit `<commit>` to point at the git commit that contains the changes to build in `specs.json`, two arbitrary named commits `<commit>..<commit>` to compile the list of changes made to `specs.json` between the two commits, or an issue number of the w3c/browser-specs repo that represents a spec suggestion and follows the expected structure. The `gh` CLI command must be available if `<what>` is an issue number.')
+  .option('-c, --commit', 'commit potential updates to a dedicated branch and switch to that branch. This option implies the `--update` option. The `git` CLI command must be available.')
+  .option('-j, --json <json>', 'link to a JSON file with additional spec properties, or a serialized JSON object directly. The option can only be set if `<what>` is a URL. The option value can also be the JSON object directly enclosed in double quotes (or without whitespaces), e.g. `"{ ""nightly"": { ""sourcePath"": ""compatibility.bs"" } }"`. A `url` property at the root level will be ignored.')
```

Comment:
shouldn't this be enclosed in single quotes instead? I'm not sure how `"{ ""nightly"": { ""sourcePath"": ""compatibility.bs"" } }"` can be parsed into valid JSON

Reply:
I wonder whether that depends on the platform :) In particular, that is how I need to write it in my Windows bash to get Commander to report the string `{ "nightly": { "sourcePath": "compatibility.bs" }  }" that I can then parse into valid JSON. Using single quotes to wrap the parameter does not work on my end.

Reply:
https://stackoverflow.com/a/15262019 suggests that this is indeed a thing in Windows shell. It's not on Unix though :)

Reply:
I adjusted the text to note that the exact syntax depends on the platform!